components/fxa-client/src/send_tab.rs

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+pub use crate::commands::send_tab::{SendTabPayload, TabData};
+use crate::{
+    commands::send_tab::{self, EncryptedSendTabPayload, PrivateSendTabKeys, PublicSendTabKeys},
+    errors::*,
+    http_client::GetDeviceResponse,
+    scopes, FirefoxAccount,
+};
+
+impl FirefoxAccount {
+    /// Ensures that the send tab command has been registered for our own device.
+    ///
+    /// **💾 This method alters the persisted account state.**
+    pub fn ensure_send_tab_registered(&mut self) -> Result<()> {
+        let own_keys: PrivateSendTabKeys =
+            match self.state.commands_data.get(send_tab::COMMAND_NAME) {
+                Some(s) => serde_json::from_str(s)?,
+                None => {
+                    let keys = PrivateSendTabKeys::from_random()?;
+                    self.state.commands_data.insert(
+                        send_tab::COMMAND_NAME.to_owned(),
+                        serde_json::to_string(&keys)?,
+                    );
+                    keys
+                }
+            };
+        let public_keys: PublicSendTabKeys = own_keys.into();
+        let kek = self.sync_keys_as_send_tab_kek()?;
```

Comment:
If `public_keys.as_command_data` and its friends accepted a `ScopedKey` argument, could this separate notion of `KeyEncryptionKey` go away and could the `sync_keys_as_send_tab_kek()` method be replaced by a call to the existing `get_scoped_key("oldsync")` method?  If so I think it would help make things easier to understand, both here and in `commands/send_tab.rs`.